<launch>
  <arg name="namespace" default="iris"/><arg name="world_name" default="basic"/>
  <arg name="paused" default="false"/><arg name="gui" default="true"/>
  <arg name="debug" default="false"/><arg name="verbose" default="false"/>
  <arg name="use_model_states_for_control" default="true"/>

  <include file="$(find clf_cbf_qp)/launch/spawn_rotors_iris.launch">
    <arg name="namespace" value="$(arg namespace)"/>
    <arg name="world_name" value="$(arg world_name)"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="verbose" value="$(arg verbose)"/>
    <arg name="x" value="31.0"/><arg name="y" value="0.0"/>
    <arg name="z" value="0.1"/><arg name="yaw" value="0.0"/>
  </include>

  <node name="clf_iris_trajectory_controller" pkg="clf_cbf_qp" type="clf_trajectory_tracking.py" output="screen">
    <param name="namespace" value="$(arg namespace)"/><param name="use_model_states" value="$(arg use_model_states_for_control)"/>
    <param name="control_rate" value="100.0"/>

    <param name="mass" value="1.5"/><param name="gravity" value="9.81"/>
    <param name="I_x" value="0.0347563"/><param name="I_y" value="0.0458929"/><param name="I_z" value="0.0977"/>

    <param name="motor_constant" value="8.54858e-06"/><param name="moment_constant" value="1.3677728e-07"/>
    <param name="max_rot_velocity" value="838.0"/>

    <param name="ellipse_a" value="31.0"/><param name="ellipse_b" value="21.0"/>
    <param name="trajectory_z" value="2.0"/><param name="trajectory_omega" value="0.05"/>
    <param name="fixed_yaw_deg" value="0.0"/>

    <param name="hover_pos_threshold" value="0.1"/><param name="hover_vel_threshold" value="0.1"/>
    <param name="hover_stabilization_secs" value="1.0"/>

    <param name="k_trajpos1" value="0.9"/>
    <param name="k_trajpos2" value="3.1"/>
    <param name="k_trajatt1" value="12.8"/>
    <param name="k_trajatt2" value="25.6"/>
    <param name="k_takepos1" value="0.25"/>
    <param name="k_takepos2" value="0.63"/>
    <param name="k_takeatt1" value="2.1"/>
    <param name="k_takeatt2" value="4.2"/>
    <param name="gravity_comp_factor" value="1.022"/>
    <param name="max_tilt_angle_deg" value="30.0"/>
    <param name="min_thrust_factor" value="0.05"/>
  </node>

  <node name="trajectory_plotter_node" pkg="clf_cbf_qp" type="trajectory_plotter.py" output="screen">
    <param name="odom_topic" value="/$(arg namespace)/ground_truth/odometry"/>
    <param name="omega_sq_topic" value="/clf_iris_trajectory_controller/control/omega_sq"/>
    <param name="thrust_topic" value="/clf_iris_trajectory_controller/control/U"/>
    <param name="state_topic" value="/clf_iris_trajectory_controller/control/state"/>

    <param name="ellipse_a" value="31.0"/><param name="ellipse_b" value="21.0"/>
    <param name="trajectory_z" value="2.0"/><param name="trajectory_omega" value="0.05"/>
    <param name="plot_save_dir" value="$(find clf_cbf_qp)/plots"/>
    <param name="min_omega" value="0.0"/><param name="max_omega" value="838.0"/>
  </node>
</launch>
